modules/core/shared/src/main/scala/lucuma/core/math/parser/Epoch.scala

```diff
@@ -25,18 +25,18 @@ trait EpochParsers {
 
   /** Parser for an `Epoch`. */
   val epoch: Parser[Epoch] =
-    (epochScheme, int <~ char('.'), intN(3))
+    (epochScheme, int.refined[Epoch.Year] <~ char('.'), intN(3))
       .mapN { (s, y, f) =>
-        s.fromMilliyears(y * 1000 + f)
+        s.fromMilliyearsUnsafe(y.value * 1000 + f)
       }
       .named("epoch")
 
   /** Parser for an `Epoch` with a default Julian schemme. */
-  val epochLenient: Parser[Epoch] =
-    (opt(epochScheme), int <~ opt(char('.')), MiscParsers.frac(3))
```

Comment:
I removed the optional scheme here because the Format that uses this Parser doesn't include the scheme in the reverseGet. This made the Format unlawful.

Reply:
👍